sw/device/silicon_creator/lib/drivers/pinmux_unittest.cc

```diff
@@ -0,0 +1,69 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/drivers/pinmux.h"
+
+#include "gtest/gtest.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/silicon_creator/lib/base/mock_abs_mmio.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "pinmux_regs.h"  // Generated.
+
+namespace pinmux_unittest {
+namespace {
+using testing::Each;
+using testing::Eq;
+using testing::Test;
+
+class PinmuxTest : public mask_rom_test::MaskRomTest {
+ protected:
+  uint32_t base_ = TOP_EARLGREY_PINMUX_AON_BASE_ADDR;
+  mask_rom_test::MockAbsMmio mmio_;
+};
+
+class InitTest : public PinmuxTest {};
+
+TEST_F(InitTest, Initialize) {
+  // Expectations for peripheral input configuration.
+  {
+    // Set to track the peripheral inputs that have already been configured.
+    auto set = std::set<top_earlgrey_pinmux_peripheral_in_t>();
+
+    // Lambda to calculate register address for peripheral input and check that
+    // it has not already been configured.
+    auto reg = [&](top_earlgrey_pinmux_peripheral_in_t index) -> uint32_t {
```

Comment:
Uh, do you think putting this in the test fixture is a better idea?

Reply:
I think you're right. Done.